### PR TITLE
Fix and refine keras tests

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Keep same dim_ordering when convert backend.
Add some more tests against different dim_orderings.

## How was this patch tested?

Jenkins.

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

**Comment by hkvision** [comment]
http://172.168.2.101:8080/view/PR-V(new)/job/BigDL-PRVN-caffe-keras/16/

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V/job/BigDL-PR-V/761/

File: pyspark/test/bigdl/test_utils.py [hunk 1]
```diff
@@ -299,11 +299,15 @@ def _do_modelTestSingleLayer(self,
...
     def __set_keras_backend(self, backend):
         if K.backend() != backend:
+            current_dim = K.image_dim_ordering()
             os.environ['KERAS_BACKEND'] = backend
             from six.moves import reload_module
             reload_module(K)
+            K.set_image_dim_ordering(current_dim)
             assert K.backend() == backend
+            assert K.image_dim_ordering() == current_dim
         if backend == "theano":
```
Comment (zhichao-li) [inline pyspark/test/bigdl/test_utils.py]: why we need this? how about remove those checking?

Comment (hkvision) [reply pyspark/test/bigdl/test_utils.py]: This is to make theano backend work with python 3.5 and 3.6

**Comment by hkvision** [comment]
jenkins new passed http://172.168.2.101:8080/view/PR-V(new)/job/BigDL-PRVN-keras/31/

**Pull request merged**